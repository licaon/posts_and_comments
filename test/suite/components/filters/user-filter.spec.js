import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import UserFilter from 'src/components/filters/user-filter';

describe('UserFilter component', () => {
  const createComponent = (props) => {
    const wrapper = shallow(
      <UserFilter
        {...props}
      />);
    return toJson(wrapper);
  };
  it('should render component correctly', (done) => {
    const component = createComponent({
      users: ['1'],
      value: '1',
      onValueChange: () => {},
    });

    expect(component).toMatchSnapshot();
    done();
  });
  it('should render component correctly without users', (done) => {
    const component = createComponent({
      onValueChange: () => {},
    });

    expect(component).toMatchSnapshot();
    done();
  });
});
