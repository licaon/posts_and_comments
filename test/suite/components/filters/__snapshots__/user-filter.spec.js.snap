// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFilter component should render component correctly 1`] = `
<SelectField
  className="md-full-width"
  defaultValue="1"
  iconChildren="arrow_drop_down"
  id="users"
  itemLabel="name"
  itemValue="id"
  keyboardMatchingTimeout={1000}
  label="Users"
  lineDirection="left"
  menuItems={
    Array [
      "1",
    ]
  }
  onChange={[Function]}
  position="tl"
  stretchList={true}
/>
`;

exports[`UserFilter component should render component correctly without users 1`] = `
<SelectField
  className="md-full-width"
  defaultValue=""
  iconChildren="arrow_drop_down"
  id="users"
  itemLabel="name"
  itemValue="id"
  keyboardMatchingTimeout={1000}
  label="Users"
  lineDirection="left"
  menuItems={Array []}
  onChange={[Function]}
  position="tl"
  stretchList={true}
/>
`;
